llvm/lib/Transforms/Utils/SimplifyCFG.cpp

```diff
@@ -4418,13 +4420,11 @@ static bool mergeConditionalStoreToAddress(
   IRBuilder<> QB(PostBB, PostBBFirst);
   QB.SetCurrentDebugLocation(PostBBFirst->getStableDebugLoc());
 
-  Value *PPred = PStore->getParent() == PTB ? PCond : QB.CreateNot(PCond);
-  Value *QPred = QStore->getParent() == QTB ? QCond : QB.CreateNot(QCond);
+  InvertPCond ^= (PStore->getParent() != PTB);
+  InvertQCond ^= (QStore->getParent() != QTB);
+  Value *const PPred = InvertPCond ? PCond : QB.CreateNot(PCond);
+  Value *const QPred = InvertQCond ? QCond : QB.CreateNot(QCond);
```

Comment:
```suggestion
  Value *const PPred = InvertPCond ? QB.CreateNot(PCond) : PCond;
  Value *const QPred = InvertQCond ? QB.CreateNot(QCond) : QCond;
```
Missed this bit before -- the logic was double-inverted previously, so it ended up being correct, and now it's wrong...

Reply:
Lol. And I followed convinced I didn't see it.

OK, clearly an argument for the readability aspect!